### PR TITLE
Fix post meta import issue

### PR DESCRIPTION
**Pull request by ouikhuan**
Fixes Automattic/dotcom-forge#1689

## Proposed changes:

* Previously some of the post meta was filtered out when importing a post, in this PR, we add some actions to process post meta after a post is created. We'll also need to add post meta into the whitelist to make it works properly.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

See issue above
## Does this pull request change what data or activity we track or use?

N/A
## Testing instructions:

* Spin up a JN site using [this branch](https://jurassic.ninja/create/?jetpack-beta&branches.jetpack=fix/meta-import&wp-debug-log)
* Connect it to your wpcom user.
* Navigate to http://calypso.lcoalhost:3000/import/${SITE_SLUG}
* Click WordPress and upload a WXR file
* Once it's finished, go to your sandbox and check if metadata gets imported correctly. ( The _oembed ones shouldn't be imported because it'll regenerate once the post is created ).
* You can use wpsh to check: `wp_${site_id}_postmeta` is the table you're looking for.

### CODE REVIEW

File: projects/packages/import/src/endpoints/class-post.php [hunk 1]
```diff
@@ -138,4 +141,63 @@ protected function extract_terms_ids( $request, $taxonomy ) {
...
+
+		$meta_keys_array = $this->filter_post_meta_keys( $metas );
+		// Adding it to the whitelist
+		Settings::update_settings( array( 'post_meta_whitelist' => $meta_keys_array ) );
+
+		if ( is_array( $metas ) ) {
+			foreach ( $metas as $meta_key => $meta_value ) {
+
+				$meta_value = maybe_unserialize( $meta_value );
+				if ( ! $creating && $meta_key === '_edit_last' ) {
```
Comment (zaerl) [inline projects/packages/import/src/endpoints/class-post.php:168]: Specific meta values cannot be saved as-is but need some help wpcom side. For example, `_edit_last` has saved the ID of the user that modified the post. On wpcom we should do this:

1. A post meta that is known to have saved an ID of a user we should change it with the remapped user
2. A post meta that is known to have saved an ID of an external entity, this ID must be changed querying the DB like it happens in the [comment endpoint](https://github.com/Automattic/jetpack/blob/fix/meta-import/projects/packages/import/src/endpoints/class-comment.php#L67-L71)

**Review by zaerl — CHANGES_REQUESTED** [review]

**Review by zaerl — APPROVED** [review]
LGTM! 🚢

**Comment by pablinos** [comment]
This PR is causing a load of extra Publicize metadata to be saved on posts. The problem is that the rest name for the meta doesn't match the key, you can see how [the name in the schema is different from the meta key registered here](https://github.com/Automattic/jetpack/blob/trunk/projects/packages/publicize/src/class-publicize-base.php#L1128-L1132)

I'm not sure how you would map those names back but also I don't think this needs to run unless an import is happening, otherwise it will interfere with how meta is saved in the block editor.

**Comment by zaerl** [comment]
> I'm not sure how you would map those names back but also I don't think this needs to run unless an import is happening, otherwise it will interfere with how meta is saved in the block editor.

This runs only when the import endpoint is called. It does not intercept core REST endpoint.

**Comment by pablinos** [comment]
> This runs only when the import endpoint is called. It does not intercept core REST endpoint.

In my tests it is running on the core REST endpoint because it's hooked into [rest_insert_post](https://github.com/Automattic/jetpack/pull/29497/files#diff-e5e8a8e1f665032f8c45fbd32219030ca29d4036132f78c8cd376697bb9fc886R43)

**Comment by ouikhuan** [comment]
> In my tests it is running on the core REST endpoint because it's hooked into [rest_insert_post](https://github.com/Automattic/jetpack/pull/29497/files#diff-e5e8a8e1f665032f8c45fbd32219030ca29d4036132f78c8cd376697bb9fc886R43)

Thanks for reporting this issue and sorry for the regression, @pablinos . In this case I'll follow up with a PR to fix this. This is intent to only run while importing a post.

**Comment by zaerl** [comment]
@ouikhuan I think that moving `add_action( "rest_insert_{$post_type}", array( $this, 'process_post_meta' ), 10, 3 );` inside `create_item` is enough to solve the issue.

**Comment by ouikhuan** [comment]
> @ouikhuan I think that moving add_action( "rest_insert_{$post_type}", array( $this, 'process_post_meta' ), 10, 3 ); inside create_item is enough to solve the issue.

True, but I guess it's no point to put it in a hook that way, I'll modify the function a bit so meet the requirement 👍

**Pull request merged**